backend/api/routes/files.py

```diff
@@ -84,16 +83,14 @@ def generate_file_hash(file: BinaryIO) -> str:
     return hasher.hexdigest()
 
 
-def save_file(file_name: str, file: BinaryIO) -> Path:
+def save_file(file: BinaryIO, file_hash: str, project_id: UUID) -> Path:
     """Saves a binary file to a specific location and returns its path."""
-    if not BackendConf.transient_storage_path.exists():
-        os.makedirs(BackendConf.transient_storage_path, exist_ok=True)
-    fpath = Path(BackendConf.transient_storage_path).joinpath(file_name)
+    BackendConf.transient_storage_path.mkdir(exist_ok=True)
```

Comment:
Given this is a _constant_, it shouldn't be any function using it's responsibility to ensure folder exists. This line should be in the BackendConf